Sources/SwiftParser/Declarations.swift

```diff
@@ -298,6 +295,13 @@ extension Parser {
       )
     )
   }
+
+  fileprivate mutating func isMissingFuncKeyword() -> Bool {
+    let isPossibleFuncIdentifier = self.at(.identifier, .wildcard)
+    let isPossibleFuncParen = self.peek(isAt: .leftParen, .binaryOperator)
+    // Treat operators specially because they're likely to be functions.
+    return (isPossibleFuncIdentifier && isPossibleFuncParen) || self.at(anyIn: Operator.self) != nil
+  }
```

Comment:
1. I would prefix this function with `at` so that it matches all the other functions that check the current token the parser is at, like `atFunctionDeclarationWithoutFuncKeyword`.
2. I would also add a doc comment.
3. And a couple suggestions to improve its performance since this becomes part of a hot loop now (the optimization is to only call `peek` if `isPossibleFuncIdentifier` is `true`
4. And finally, why do we need to check for `_` as the func identifier? I don’t think that `_` is a valid function name.
5. And we shouldn’t just be checking for a binary operator. It makes more sense to check for the `<` that starts a generic parameter clause here.

```suggestion
  /// Returns `true` if it looks like the parser is positioned at a function declaration that’s missing the `func` keyword.
  fileprivate mutating func atFunctionDeclarationWithoutFuncKeyword() -> Bool {
    if self.at(.identifier) && (self.peek(isAt: .leftParen) || self.peek().tokenText.hasPrefix("<")) {
      return true
    }
    if self.at(anyIn: Operator.self) != nil {
      return true
    }
    return false
  }
```

Reply:
Not sure about number 4. It was already there.